### PR TITLE
feat: use PubSub API directly from libp2p

### PR DESCRIPTION
**Pull request by diasdavid**

### CODE REVIEW

**Comment by diasdavid** [comment]
Done

**Comment by diasdavid** [comment]
Had to fix some linting issues that were merged in Master (somehow escaped the check, shame on me for not triple checking.)

**Comment by diasdavid** [comment]
image

CI is green \o/ Merging as this solves also some issues in Master that are preventing other PRs

File: src/core/components/bootstrap.js [hunk 1]
```diff
@@ -1,9 +1,24 @@
...
 const defaultNodes = require('../runtime/config-nodejs.json').Bootstrap
-const MultiAddr = require('multiaddr')
+const Multiaddr = require('multiaddr')
 const promisify = require('promisify-es6')
 
+function isValid (ma) {
+  if (typeof ma === 'string') {
+    try {
+      ma = new Multiaddr(ma)
+      return Boolean(ma)
```
Comment (vmx) [inline src/core/components/bootstrap.js]: This idea comes from type checking discussion. It would be cool if `isValid()` functions would return `false` if invalid and return a valid object whatever got checked (in this case a `Multiaddr`).

That way you could also use `isValid()` to make sure you always get a valid `Multiaddr` object, no matter if it was already one or of if it was a string.

File: src/core/components/libp2p.js [hunk 2]
```diff
@@ -50,9 +51,7 @@ module.exports = function libp2p (self) {
         })
 
         self._libp2pNode.start((err) => {
-          if (err) {
-            return callback(err)
-          }
+          if (err) { return callback(err) }
```
Comment (vmx) [inline src/core/components/libp2p.js]: Nitpick: I prefer not having unrelated formatting changes together with real code changes.

File: test/core/bitswap.spec.js [hunk 3]
```diff
@@ -16,10 +16,9 @@ const isNode = require('detect-node')
 const multihashing = require('multihashing-async')
 const CID = require('cids')
 
-const DaemonFactory = require('ipfsd-ctl')
-const df = DaemonFactory.create({ type: 'js' })
-
-const dfProc = DaemonFactory.create({ type: 'proc' })
+const IPFSFactory = require('ipfsd-ctl')
+const fDaemon = IPFSFactory.create({ type: 'js' })
```
Comment (vmx) [inline test/core/bitswap.spec.js]: I find `fDaemon` a strange, hard to read, uncommon name. What about just `Deamon` or `jsDeamon`.

File: test/core/bitswap.spec.js [hunk 4]
```diff
@@ -16,10 +16,9 @@ const isNode = require('detect-node')
...
+const fInProc = IPFSFactory.create({ type: 'proc' })
```
Comment (vmx) [inline test/core/bitswap.spec.js]: I find `fInProc` a strange, hard to read, meaningless, uncommon name. What about just `InProc` or `InProcDeamon`.

**Comment by vmx** [comment]
Oh, I just saw that it got already merged :) Well, then see my comments as FYI and fill free to do a new PR if you think it's worth it.

File: test/core/bitswap.spec.js (hunk 3 above)
Comment (diasdavid) [reply test/core/bitswap.spec.js]: It's a Factory. I could call it factoryDaemon if that makes more sense for you

File: src/core/components/bootstrap.js (hunk 1 above)
Comment (diasdavid) [reply src/core/components/bootstrap.js]: This got upgraded in https://github.com/ipfs/js-ipfs/pull/1227

File: test/core/bitswap.spec.js (hunk 4 above)
Comment (diasdavid) [reply test/core/bitswap.spec.js]: Same as https://github.com/ipfs/js-ipfs/pull/1215#discussion_r169066092. I can use the full word if that works better :)

File: test/core/bitswap.spec.js (hunk 3 above)
Comment (vmx) [reply test/core/bitswap.spec.js]: Does it matter that's a factory? If you really want to know, I think it's apparent from `IPFSFactor.create()` that it is a factory.
This don't find this super important, so if you don't find it worth the change, feel free to leave it as it currently is.
